assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -142,6 +164,74 @@ export const actions = {
 			},
 			type: REGISTER_NOTIFICATION,
 		};
+
+		const registry = yield commonActions.getRegistry();
+
+		// If no view contexts were provided, we should instead do a comparison
+		// with the "top"/"visible" notification in the queue for this
+		// notification's `groupID` to see if the newly-registered notification
+		// should be added to the "top"/"visible position" in the queue.
+		//
+		// This is the usual route for "ad-hoc" notifications that are registered
+		// after initial page load, such as Setup Success notifications.
+		if ( ! viewContexts?.length ) {
+			// To do this, we'll add the notification to the queue immediately, which
+			// will insert it in the right position based on its priority and other
+			// criteria (handled by the reducer for this action).
+			registry
+				.dispatch( CORE_NOTIFICATIONS )
+				.addNotificationToQueue( id );
+
+			return;
+		}
+
+		// If view contexts were provided, we need to repopulate the queues because
+		// because the data store doesn’t know the current `viewContext` and thus
+		// can’t figure out if the ones specified are valid/active.
```

Comment:
Since we don't store (and shouldn't need to store) separate queues for each `viewContext` we could store the `viewContext` that the queue was populated (or is populating) for in state. We could call it the "queue context". Then we could use this to conditionally reset the queue when the resolver runs again since we should only need to clear it if the context has changed.